### DISCUSSION TITLE
How to avoid adding single quote

### DISCUSSION THREAD

**Discussion by fmuyassarov** [pelletier/go-toml#904] [Questions] [answered]
I'm unmarshalling a TOML file, then modifying its content and writing it back to the TOML file. During the modifications I'm only adding a new field  `crio.nri` and according to go-toml v2 it is getting quoted with single quotes. However, this is not desired in my case and I would like to avoid adding single/double quotes. Does go-toml v2 support it natively? 

Before
```
[crio]
[crio.api]
[crio.image]
[crio.metrics]
```
After modifying with go-toml v2
```
[crio]
[crio.api]
[crio.image]
[crio.metrics]
# This is new table I'm adding and I would like to avoid ''
['crio.nri']
```

**Comment by fmuyassarov** [comment]
ping @pelletier

**Comment by pelletier** [comment]
The single quotes are here to denote that `crio.nri` is the name of the TOML table, as opposed to the key `nri` inside the `crio` table which sounds like what you want to get. See the [table spec](https://toml.io/en/v1.0.0#table). Would you mind sharing the code you're using to generate this output to know whether it's a bug?

**Comment by fmuyassarov** [comment]
@pelletier thanks for the prompt response. Below is the code block:
- [reading](https://github.com/fmuyassarov/nri-plugins/blob/ba694bc43737bf28f32f4c9fed1177eea73bffac/cmd/config-manager/main.go#L108) a file in TOML format 
- [modifying](https://github.com/fmuyassarov/nri-plugins/blob/crio-nri/cmd/config-manager/main.go#L141) (adding a table) to unmarshalled TOML
- [writing](https://github.com/fmuyassarov/nri-plugins/blob/ba694bc43737bf28f32f4c9fed1177eea73bffac/cmd/config-manager/main.go#L86) back edited table to the TOML file

I think what you want to see is the last point [writing](https://github.com/fmuyassarov/nri-plugins/blob/ba694bc43737bf28f32f4c9fed1177eea73bffac/cmd/config-manager/main.go#L86)

**Reply by pelletier** [reply, marked as answer]
The issue is in the modification. The `crioNriTable` key is defined as `"crio.nri"`. The whole string is the key. You'd need to do it in two steps. For example (checks omitted for brevity) :

```go
crio, ok := config["crio"].(map[string]interface{})
nri, ok := crio["nri"].(map[string]interface{})
nri["enable_nri"] = true
```

**Reply by fmuyassarov** [reply]
Sorry for late response. That indeed was the culprit and your example works nicely. Thank you very much @pelletier !
